pkg/api/mlflow/dao/repositories/metric.go

```diff
@@ -85,8 +85,17 @@ func (r MetricRepository) CreateBatch(
 		lastIters[lastMetric.Key] = lastMetric.LastIter
 	}
 
+	contextMap := make(map[string]*models.Context)
 	latestMetrics := make(map[string]models.LatestMetric)
 	for n, metric := range metrics {
+		if metric.Context != nil {
+			hash := getJsonHash(metric.Context.Json)
+			if contextMap[hash] == nil {
+				contextMap[hash] = metric.Context
+			} else {
+				metrics[n].Context = contextMap[hash]
+			}
+		}
```

Comment:
Because yes the context's content is the same but with a different pointer so in this way every metric with the same context will have the same reference